### DISCUSSION TITLE
What is "Knowledge" and how to use it?

### DISCUSSION THREAD

**Discussion by tosho** [fmaclen/hollama#183] [General]
Basically the title. 
I have no idea what that is and how to use it. Can someone explain it to me?

**Comment by fmaclen** [comment]
The idea behind **Knowledge** is that you can re-use the same piece of content on multiple sessions. Similar to the way [Claude Projects](https://www.anthropic.com/news/projects) work.

At the moment the feature is really limited. You can only use **Knowledge** as the "System prompt" when you create a **New Session**, the content inside the knowledge will be passed together with your prompt to the AI when you click **Run**.

In the future we plan to expand on this feature so you can use Knowledge not only as a System prompt but as "additional context" on every prompt. There are also plans to have multiple "types" of Knowledge such as: images, links to websites, files.

Here's some screenshots of what we have in mind, but nothing is set in stone yet:
https://github.com/fmaclen/hollama/issues/116#issuecomment-2251424660

**Comment by tosho** [comment]
I tried now. It's more like instructions? "Act as Assistant, Friend, Doctor" kinda?
How will that works with more information like the examples if they're inserted in every prompt? Wouldn't take affect responses?
On another note (I'm not familiar) but can LLM's take new data into consideration like for example they're trained on a X_version library/framework/language from some time ago but there is a new one and you want to them to have that new information and refactor their response with that. Or that requires new training?

**Comment by fmaclen** [comment]
> It's more like instructions? "Act as Assistant, Friend, Doctor" kinda?

Correct, "System prompts" are instructions you can give the model to steer the conversation in any particular way.

> How will that works with more information like the examples if they're inserted in every prompt? Wouldn't take affect responses?

The examples are showing additional content you can send with your prompt as "context", not "System prompts" which typically only sent with the first prompt.

> On another note (I'm not familiar) but can LLM's take new data into consideration like for example they're trained on a X_version library/framework/language from some time ago but there is a new one and you want to them to have that new information and refactor their response with that. Or that requires new training?

There are [multiple techniques](https://ollama.com/blog/embedding-models) to get an LLM to use "up-to-date" knowledge to answer questions, without having to re-train a model.

The simplest way, which is what we are hoping to achieve with the future version of Knowledge, is to just include a bunch of text together with your prompt that the LLM can interprete and answer based on it, even if it wasn't trained on that content.

The only limitation there is that there is a "context window" limit each model has, that will determine how much extra "text" you can submit with your prompt. You can find the sizes of each model in [Ollama's library](https://ollama.com/library/mistral-nemo).
